services/venice-controller/src/main/java/com/linkedin/venice/pushmonitor/AbstractPushMonitor.java

```diff
@@ -1120,15 +1123,37 @@ private void updateStoreVersionStatus(String storeName, int versionNumber, Versi
                     storeName,
                     versionNumber));
           }
-          if (version.isVersionSwapDeferred()) {
+
+          /**
+           * Switch to the new version if:
+           * 1.deferred version swap is not enabled and it is not a target region push w/ deferred version swap
+           * 2.target region push w/ deferred swap is enabled and the current region matches the target region
+           *
+           * Do not switch to the new version now if:
+           * 1.deferred version swap is enabled (it will be manually swapped at a later date)
+           * 2.target region push is enabled and the current region does NOT match the target region (it will be swapped
+           *  after targetSwapRegionWaitTime passes)
+           */
+          Set<String> targetRegions = RegionUtils.parseRegionsFilterList(version.getTargetSwapRegion());
+          boolean isValidTargetRegion = targetRegions.contains(regionName);
+          boolean isTargetRegionPushWithDeferredSwap = version.isVersionSwapDeferred() && isValidTargetRegion;
```

Comment:
maybe combine the 2 boolean into one.
`boolean performVersionSwap = !version.isVersionSwapDeferred() ||  isValidTargetRegion;`